driver-core/src/test/functional/com/mongodb/client/CommandMonitoringTestHelper.java

```diff
@@ -187,9 +187,12 @@ private static CommandSucceededEvent massageActualCommandSucceededEvent(final Co
             if (response.containsKey("writeErrors")) {
                 for (BsonValue bsonValue : response.getArray("writeErrors")) {
                     BsonDocument cur = bsonValue.asDocument();
-                    cur.put("code", new BsonInt32(42));
-                    cur.put("errmsg", new BsonString(""));
-                    cur.remove("codeName");
+                    BsonDocument newWriteErrorDocument =
+                            new BsonDocument().append("index", cur.get("index"))
+                                    .append("code", new BsonInt32(42))
+                                    .append("errmsg", new BsonString(""));
+                    cur.clear();
```

Comment:
Could also just remove the specific fields that are being added in MongoDB 4.2 (e.g. keyPattern), but server team has not been able to guarantee that there won't be others added in the future.